src/main/scala/com/sagacify/sonar/scala/ScalaSensor.scala

```diff
@@ -1,35 +1,52 @@
 package com.sagacify.sonar.scala
 
-import scala.io.Source
-import scala.collection.JavaConverters._
-
-import org.sonar.api.batch.fs.FileSystem
-import org.sonar.api.batch.Sensor
-import org.sonar.api.batch.SensorContext
+import org.sonar.api.batch.sensor.{Sensor, SensorContext, SensorDescriptor}
 import org.sonar.api.measures.{CoreMetrics => CM}
-import org.sonar.api.resources.Project
 
-class ScalaSensor(scala: Scala, fs: FileSystem) extends Sensor {
+import scala.collection.JavaConverters._
+import scala.io.Source
 
-  def shouldExecuteOnProject(project: Project): Boolean =
-    fs.hasFiles(fs.predicates.hasLanguage(scala.getKey))
+class ScalaSensor(scala: Scala) extends Sensor {
 
-  def analyse(project: Project, context: SensorContext): Unit = {
-    val charset = fs.encoding.toString
-    val version = "2.11.8"
+  override def execute(context: SensorContext): Unit = {
+    val charset = context.fileSystem().encoding.toString
+    val versionProperty = context.config().get("sonar.scala.source")
+    val version =
+      if (context.config().get("sonar.scala.source").isPresent) versionProperty.get() else "2.11.11"
```

Comment:
`sonar.scala.source` might be a little bit confusing given that there is already `sonar.sources` key in [SonarQube](https://docs.sonarqube.org/display/SONAR/Analysis+Parameters#AnalysisParameters-ProjectConfiguration).

Since this is related to Scala version, I'd probably name it `sonar.scala.version`.

Reply:
Could you default it to the [`Scala_2_11`](https://github.com/scala-ide/scalariform/blob/master/scalariform/src/main/scala/scalariform/ScalaVersions.scala#L38) constant from Scaliform?

Reply:
Done